prefect_dbt/cloud/jobs.py

```diff
@@ -397,17 +404,22 @@ async def _build_trigger_job_run_options(
                     step=run_step["index"],
                 )
                 run_artifact = await run_artifact_future.result()
-                run_results = run_artifact["results"]
             except DbtCloudGetRunArtifactFailed:
                 steps_override.append(command)
             else:
-                run_nodes = " ".join(
+                run_results = run_artifact["results"]
+                run_nodes = [
                     run_result["unique_id"].split(".")[2]
                     for run_result in run_results
                     # "fail" here instead of "cancelled"
                     if run_result["status"] in ("error", "skipped", "fail")
+                ]
+
+                # e.g. dbt --experimental_parser, build, --vars '{"env": "prod"}'
+                dbt_global_args, exe_command, exe_args = command.partition(exe_command)
+                modified_command = (
```

Comment:
I do like this approach more as it doesn't require parsing arguments.  However, I think your modified command could contain multiple `--select` or `-s` arguments.  For example, if my command that fails is:
```
dbt --fail-fast run -s bad_model --vars '{"key": "value"}'
```
The components you would use to create a modified command would be:
- `dbt_global_args` = `dbt --fail-fast `
- `exe_command` = `run`
- `exe_args` = ` -s bad_model --vars '{"key": "value"}'`

Which would then create:
```
dbt --fail-fast run --select bad_model -s bad_model --vars '{"key": "value"}'
```

Reply:
That's a good point. I think I'll search for `-s` and `--select`, then append args if it exists. If not use the current method.